specification/eventgrid/Azure.Messaging.EventGrid/main.tsp

```diff
@@ -0,0 +1,236 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-autorest";
+import "@azure-tools/typespec-azure-core";
+
+enum ServiceApiVersions {
+  v2022_05_01: "2023-06-01-preview"
+}
+
+@useAuth(
+  ApiKeyAuth<ApiKeyLocation.header, "aeg-sas-key"> | OAuth2Auth<[{
```

Comment:
While testing the service, what we see is that the only autg support is:
Authorisation: SharedAccessKey

Reply:
Confirmed with the team and you are right. I updated it

@useAuth(
  ApiKeyAuth<ApiKeyLocation.header, "SharedAccessKey"> | OAuth2Auth<[{
    type: OAuth2FlowType.implicit,
    authorizationUrl: "https://login.microsoftonline.com/common/oauth2/v2.0/authorize",
    scopes: ["https://eventgrid.azure.net/.default"],
  }]>
)

Reply:
@ahamad-MS This is not correct. The header name is `Authorisation`, and the value is `SharedAccessKey \<insert your key\>`. Your TypeSpec says that the header name is `SharedAccessKey`, and that we can put the key as value, which is not what we saw.

My question for the stewardship is wider though: is that an ok way to auth? The fact that we cannot describe this in TypeSpec today, is likely an indicator that it's not following Azure guidelines recommendation on auth.